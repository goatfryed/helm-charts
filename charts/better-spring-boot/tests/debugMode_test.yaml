suite: debug mode
templates:
  - deployment.yaml
tests:
  - it: replaces run command
    values:
      - "values/values.base.yaml"
    set:
      debugDeployment:
        enabled: true
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec