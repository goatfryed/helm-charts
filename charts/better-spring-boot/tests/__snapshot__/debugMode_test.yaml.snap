adds health checks only if explicitly activated:
  1: |
    containers:
      - command:
          - sh
          - -c
          - trap "exit 0" SIGTERM SIGINT; tail -f /dev/null & wait
        env:
          - name: SPRING_PROFILES_ACTIVE
            value: ""
        image: goatfryed/spring-boot-echo:latest
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: http
        name: spring-boot
        ports:
          - containerPort: 80
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: http
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
            readOnly: true
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-app-config
        name: helm-application-properties
overrides profiles:
  1: |
    containers:
      - env:
          - name: SPRING_PROFILES_ACTIVE
            value: debug, profile2
        image: goatfryed/spring-boot-echo:latest
        imagePullPolicy: IfNotPresent
        name: spring-boot
        ports:
          - containerPort: 80
            name: http
            protocol: TCP
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
            readOnly: true
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-app-config
        name: helm-application-properties
overrides run command:
  1: |
    containers:
      - command:
          - sh
          - -c
          - trap "exit 0" SIGTERM SIGINT; tail -f /dev/null & wait
        env:
          - name: SPRING_PROFILES_ACTIVE
            value: ""
        image: goatfryed/spring-boot-echo:latest
        imagePullPolicy: IfNotPresent
        name: spring-boot
        ports:
          - containerPort: 80
            name: http
            protocol: TCP
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
            readOnly: true
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-app-config
        name: helm-application-properties
